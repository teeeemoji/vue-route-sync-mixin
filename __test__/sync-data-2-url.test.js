/**
 * Summary.
 *  main test file.
 *
 * Description
 *  This is the main test file
 *
 * @file   This main test file for class vue-route-sync-mixin.
 * @author teeeemoji.
 * @since 2020-1-5
 */
import {createLocalVue, mount} from '@vue/test-utils'
import VueRouter from 'vue-router'
import Comp from './test-material/component.vue'
import App from './test-material/app.vue'

describe('initial data from url', function () {
  let wrapper = null
  let comp = null
  beforeAll(() => {
    const localVue = createLocalVue()
    const routes = [{path: '/', component: Comp}]
    const router = new VueRouter({
      routes
    })
    localVue.use(VueRouter)
    wrapper = mount(App, {
      localVue,
      router,
      attachToDocument: true
    })
    comp = wrapper.vm.$refs.comp
    Object.assign(comp, {
      pagination: {
        pageSize: 1000,
        pageNum: 1000
      },
      name: 'ztabcd',
      smoke: true,
      departments: [1, 2, 3, 'four'],
      createDate: {
        start: '1067385600000',
        end: '322790400000'
      },
      und: [1, 2, 3, 4, 5, 6, 7]
    })
  })

  afterAll(() => {
    wrapper.destroy()
    wrapper = null
    comp = null
  })

  it('check if url is correct', function (done) {
    setTimeout(() => {
      expect(comp.$route.query).toMatchSnapshot()
      done()
    }, 500)
  })
})
