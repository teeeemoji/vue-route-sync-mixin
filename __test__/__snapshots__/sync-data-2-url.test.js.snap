// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initial data from url check if url is correct 1`] = `
Object {
  "c": "3__{\\"start\\":\\"1067385600000\\",\\"end\\":\\"322790400000\\"}",
  "d": "3__[1,2,3,\\"four\\"]",
  "n": "2__ztabcd",
  "pn": "1__1000",
  "ps": "1__1000",
  "s": "5__true",
  "u": "3__[1,2,3,4,5,6,7]",
}
`;
